src/lib/opencv.ts

```diff
@@ -0,0 +1,174 @@
+    // eslint-disable-next-line
+    this.content += chunk.toString();
+    callback();
+  }
+
+  getContent(): string {
+    return this.content;
+  }
+}
+
+export class OpenCvLib extends RemoteLib {
+  private dockerInstance: Docker|null = null;
+
+  async executeRemoteCall(callName: string, args: any[]): Promise<any> {
+    if (-1 === ALLOWED_CALLS.indexOf(callName)) {
+      throw new TypeError(`Unauthorized OpenCV call name: ${callName}`);
+    }
+
+    if (!this.dockerInstance) {
+      this.initDockerInstance();
+    }
+
+    const dockerInstance = this.dockerInstance as Docker;
+
+    const images = await dockerInstance.listImages();
+    if (!images.find(image => image.RepoTags && image.RepoTags.includes(OPENCV_IMAGE))) {
+      await new Promise(resolve => {
+        void dockerInstance.pull(OPENCV_IMAGE, (err: any, stream: NodeJS.ReadableStream) => {
+          dockerInstance.modem.followProgress(stream, resolve);
+        });
+      });
+    }
+
+    const resultImageName = imageCache.generateIdentifier();
+
+    const program = `import cv2
+result = cv2.${callName}(${await this.formatArguments(args)})
+cv2.imwrite('${resultImageName}', result)`;
+
+    fs.writeFileSync(`${CACHE_FOLDER}/app.py`, program);
+
+    const outputStream = new EchoStream();
+    const cacheFolder = path.join(__dirname, '../../', CACHE_FOLDER);
+
+    const data = await dockerInstance.run(OPENCV_IMAGE, ['python3', 'app.py'], outputStream, {
+      HostConfig: {
+        Binds: [`${cacheFolder}:/opt/build`],
+        AutoRemove: true,
+      },
+    }) as {StatusCode: number}[];
+
+    if (0 !== data[0].StatusCode) {
+      throw new Error(outputStream.getContent().trim());
+    }
+
+    return {
+      fileType: 'image',
+      fileUrl: '/image-cache/' + resultImageName,
+    };
+  }
+
+  async formatArguments(args: any[]): Promise<string> {
+    const argsString = [];
+    for (const arg of args) {
+      argsString.push(await this.convertArgument(arg));
+    }
+
+    return `${argsString.join(', ')}`;
+  }
+
+  async convertArgument(arg: any): Promise<string> {
+    if ('object' === typeof arg && isFileArg(arg)) {
+      const fileName = arg.fileUrl.split('/').pop() ?? '';
+      if (!fileName.match(/^[a-zA-Z0-9-]+\.[a-z]+$/)) {
+        throw new TypeError(`File name "${fileName}" does not match the required pattern`);
+      }
+
+      return `cv2.imread("${fileName ?? ''}")`;
+    }
+    if ('string' === typeof arg) {
+      if (this.isImageUrl(arg)) {
+        const imagePath = imageCache.generateIdentifier();
+        await this.downloadImage(arg, imageCache.getCachePath(imagePath));
+
+        return `"${imagePath}"`;
+      }
+
+      // eslint-disable-next-line
+      arg = arg.replace(/"/g, '\\\"');
+
+      return `"${String(arg)}"`;
+    }
+
+    if (Array.isArray(arg)) {
+      const convertedArgs = [];
+      for (const element of arg) {
+        convertedArgs.push(await this.convertArgument(element));
+      }
```

Comment:
I would have written it a `arg.map(e => ....)` (something like `await Promise.all(arg.map(e => this.convertArgument(e))`)